### DISCUSSION TITLE
ModuleNotFoundError: No module named 'udls'

### DISCUSSION THREAD

**Discussion by sophistrionics** [acids-ircam/RAVE#132] [Q&A]
Trying to install Rave. Getting the below when running train_rave.py. Had a number of similar errors before, each of which resolved by using "PIP Install _____." No luck with this one. Feeling like I made a foundational mistake earlier in the set-up. Any advice would be appreciated. Happy to provide any info needed to help with an answer. (FYI - searched discussion for "udls" and got no results.)

> Sophistrionics-MacBook-Pro:rave Sophistrionics $ python train_rave.py -c small --name test_3 --wav /documents/music/slinter/bounces --preprocessed slinter/test_3/rave 
Traceback (most recent call last):
  File "/Users/Sophistrionics/Documents/GitHub/RAVE/train_rave.py", line 8, in <module>
    from udls import SimpleDataset, simple_audio_preprocess
ModuleNotFoundError: No module named 'udls'

**Comment by caillonantoine** [comment]
You should install everything using "pip install -r requirements.txt" !

**Reply by sophistrionics** [reply]
Thanks Antoine, I very much appreciate your reply. I previously started by using "pip install -r requirements.txt". After hitting return, I received received the error quoted below. I then found your cached_conv git repository and tried installing that. From there, I kept receiving ModuleNotFoundErrors for different items, but was able to proceed past them using pip install until I hit the 'udls' error.  I cannot tell if I did something wrong _before_ I used "pip install -r requirements.txt"  (i.e., there is something about my computer or system that does not meet the requirements for Rave) or whether I should have taken a different action at the point I received the "cached_conv" error quoted below.

> Collecting GPUtil==1.4.0
  Using cached GPUtil-1.4.0.tar.gz (5.5 kB)
ERROR: Could not find a version that satisfies the requirement cached-conv (unavailable) (from versions: none)
ERROR: No matching distribution found for cached-conv (unavailable)

**Reply by caillonantoine** [reply]
It looks like you are trying to train a model on macos, which is not recommended for now. [Look at this guide](https://github.com/acids-ircam/RAVE/blob/master/docs/training_setup.md) to get details on how to setup a training machine !
